core/src/main/java/hudson/model/Label.java

```diff
@@ -389,9 +389,8 @@ private String toString(Collection<? extends ModelObject> model) {
     @Exported
     public List<AbstractProject> getTiedJobs() {
         List<AbstractProject> r = new ArrayList<>();
-        for (AbstractProject<?,?> p : Jenkins.get().allItems(AbstractProject.class)) {
-            if(p instanceof TopLevelItem && this.equals(p.getAssignedLabel()))
-                r.add(p);
+        for (AbstractProject<?,?> p : Jenkins.get().allItems(AbstractProject.class, i -> i instanceof TopLevelItem && this.equals(i.getAssignedLabel()))) {
```

Comment:
I don't think it improves performance that much, but you could stream the result from allItems() and collect it into a list, avoiding the for loop.

Reply:
Rewrote this bit to use streams